### ISSUE TITLE
docker error: exec: "ssh": executable file not found in $PATH

### ISSUE THREAD

**Issue by MarvinJWendt** [garethgeorge/backrest#156]
Running backrest in docker and connecting to an sftp repository results in the following error: 

```
Operation failed: [unknown] failed to init repo: init failed: command "/data/restic-0.16.4 init --json -o sftp.args=-oBatchMode=yes" failed: exit status 1 Process STDOUT: Fatal: create repository at sftp:REDACTED failed: exec: "ssh": executable file not found in $PATH
```

**Comment by MarvinJWendt** [comment]
I just saw this comment:

https://github.com/garethgeorge/backrest/issues/50#issuecomment-1999604891

Seems that this is fixed in `garethgeorge/backrest:v0.14.0-alpine` but not in `garethgeorge/backrest:latest` .

Could you update the `latest` tag too? :)

**Comment by MarvinJWendt** [comment]
### Just my two cents:

`ssh` is needed for `sftp` to work, without it - restic cannot back up to `sftp` repositories.
You could argue, that the base configuration should be "plug and play" and contain all necessary dependencies.

Usually, the `latest` tag is the biggest tag, which contains everything needed.
Let's take the golang `latest` tag as an example (~2 GB):
<img width="1254" alt="image" src="https://github.com/garethgeorge/backrest/assets/31022056/e0d6bbc4-1673-4958-adf5-8e4c313b1fa7">

`alpine` is usually the smallest image (as alpine has a small base image, it makes sense to make use of the small size and to not install everything into the image) Let's take the golang `alpine` tag as an example (60 MB):
<img width="1267" alt="image" src="https://github.com/garethgeorge/backrest/assets/31022056/df3a66d3-de76-4784-bb8c-5ebf082fbcbb">

If the image also provides a `scratch` tag, that is most likely the smallest. Here, there is usually nothing added except the binary. It's "from scratch" after all 😉 

The same thing can be observed on most other docker images:
- https://hub.docker.com/_/python/tags
- https://hub.docker.com/_/node/tags
- https://hub.docker.com/_/rust/tags
- etc...

Having all dependencies in the `latest` tag makes the image easy to use and the user is not required to read the docs.

My current experience with setting up backrest with docker was like this:

1. "Cool, they have a docker image!"
2. *copies docker compose example to server*
3. *starts container*
4. *trying to add SFTP repo*
5. **error**
6. "Hmm, looks like there is no ssh in the container"
7. *searches issues on GitHub*
8. "Maybe I need to write a custom image and add ssh there?"
9. *searches issues on GitHub more intensively*
10. "Ah, it's fixed in alpine (but alpine implies that there are fewer dependencies than in `latest`?)"
11. "Okay, let's use alpine for now"

With all dependencies in the `latest` tag, the process for new users would look like this:

1. "Cool, they have a docker image!"
2. *copies docker compose example to server*
3. *starts container*
4. *trying to add SFTP repo*
5. "Nice, it works"

(I hope you enjoyed my inner monologue 😆)


Also, some nitpicking:

`latest-alpine` could probably be renamed to `alpine`, as a missing version usually indicates that it's the latest image :)

**Comment by MarvinJWendt** [comment]
I just saw that your `latest` tag is build from scratch, so I guess you want to keep it really small 😄 

So maybe you could make `alpine` the `latest` tag and introduce `:scratch` or `latest-scratch`?

IMO the default configuration should "just work", people who want to keep the size of their services minimal are more likely to go through the docs and be willing to customize their images.

**Comment by garethgeorge** [comment]
Thanks!! This type of feedback is _very_ useful as an OSS author. Your write up is a great summary of a user's expectations & also OSS norms. 

I definitely find myself wishing I had given a bit more thought to image labeling when first setting up my releases-- some of how things are the way they are is legacy. The scratch image predated the alpine image by a good margin.

I think you're right though that it's worth getting consistent with other published projects & I increasingly can see that most users will care far less about the 60 MB-100 MB of disk space added by the alpine image than any amount of time spent debugging or finding depedencies!

I think this is mostly a summary of what you've proposed, but to be explicit about it -- it sounds like a safe migration plan will require me to keep all existing images. I think the safe plan will be to provide the following tags

 * `latest` - convert this to an alpine image. It may break some folks if they're doing docker builds on top of the scratch image but it's probably fine for the vast majority.
 * `scratch` - this will be the same as `latest` today, will be the scratch derived minimal image.
 * `latest-alpine` - will be synonymous with `latest`, preserved for backwards compatibility.

**Comment by MarvinJWendt** [comment]
> I increasingly can see that most users will care far less about the 60 MB-100 MB of disk space added by the alpine image than any amount of time spent debugging or finding depedencies!

Exactly, when I started to publish OSS docker images, I was also intrigued by the smallest image size possible. I deleted debug information in the Go binary and compressed it using third party tools. I also used the scratch image, which got my total image size to <5 MB. While it was a nice flex to have a full-stack App running on a ~3-4 MB docker image, I noticed, that it's not worth the effort. I still use `scratch` for anything that has no dependencies - `alpine` for everything else (Go related).

> `latest` - convert this to an alpine image. It may break some folks if they're doing docker builds on top of the scratch image but it's probably fine for the vast majority.

I agree, anyone using `latest` is risking a breaking change. A `docker pull` would also do major updates of the image (and possibly destroy a whole underlying database (*I learned the hard way years ago*))
